consensus/polybft/bridge_batch.go

```diff
@@ -0,0 +1,201 @@
+package polybft
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/0xPolygon/polygon-edge/bls"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/crypto"
+	"github.com/0xPolygon/polygon-edge/state/runtime/precompiled"
+	"github.com/0xPolygon/polygon-edge/types"
+	merkle "github.com/Ethernal-Tech/merkle-tree"
+)
+
+// PendingBridgeBatch holds merkle trie of bridge transactions accompanied by epoch number
+type PendingBridgeBatch struct {
+	*contractsapi.BridgeMessageBatch
+	Epoch uint64
+}
+
+// NewPendingBridgeBatch creates a new commitment object
+func NewPendingBridgeBatch(epoch uint64,
+	bridgeEvents []*contractsapi.BridgeMsgEvent) (*PendingBridgeBatch, error) {
+
+	messages := make([]*contractsapi.BridgeMessage, len(bridgeEvents))
+
+	for i, bridgeEvent := range bridgeEvents {
+		messages[i] = &contractsapi.BridgeMessage{
+			ID:                 bridgeEvent.ID,
+			Sender:             bridgeEvent.Sender,
+			Receiver:           bridgeEvent.Receiver,
+			Payload:            bridgeEvent.Data,
+			SourceChainID:      bridgeEvent.SourceChainID,
+			DestinationChainID: bridgeEvent.DestinationChainID}
+	}
+
+	return &PendingBridgeBatch{
+		BridgeMessageBatch: &contractsapi.BridgeMessageBatch{
+			Messages:           messages,
+			DestinationChainID: bridgeEvents[0].DestinationChainID,
+			SourceChainID:      bridgeEvents[0].SourceChainID},
+		Epoch: epoch,
+	}, nil
+}
+
+// Hash calculates hash value for commitment object.
+func (cm *PendingBridgeBatch) Hash() (types.Hash, error) {
+	data, err := cm.BridgeMessageBatch.EncodeAbi()
+	if err != nil {
+		return types.Hash{}, err
+	}
+
+	return crypto.Keccak256Hash(data), nil
+}
+
+var _ contractsapi.StateTransactionInput = &BridgeBatchSigned{}
+
+// BridgeBatchSigned encapsulates commitment message with aggregated signatures
+type BridgeBatchSigned struct {
+	MessageBatch *contractsapi.BridgeMessageBatch
+	AggSignature Signature
+	PublicKeys   [][]byte
+}
+
+// Hash calculates hash value for commitment object.
+func (cm *BridgeBatchSigned) Hash() (types.Hash, error) {
```

Comment:
Change this in every function:

```suggestion
func (bbs *BridgeBatchSigned) Hash() (types.Hash, error) {
```